packages/server/src/arpa_reporter/services/persist-upload.js

```diff
@@ -19,10 +21,15 @@ const uploadFSName = (upload) => {
   return path.join(UPLOAD_DIR, filename)
 }
 
+const jsonFSName = (upload) => {
+  const filename = `${upload.id}.json`
+  return path.join(TEMP_DIR, filename)
+}
```

Comment:
I guess the only challenge with this approach would be that the lookup logic gets a little more complicated.  I think we might have to add some columns to the upload database to maintain the cache date, so we know where to locate the parsed upload.

What issues do you see arising from a large, flat directory?

Reply:
My thought was that by using the `createdAt` field on the object, we could still avoid adding any additional database columns, since `createdAt` is (presumably) a value that won't ever change. 

The preference for avoiding a single large directory is simply because I think it helps avoid some logistical challenges- it makes manually navigating around in debugging scenarios easier, and avoids performance issues (for example, listing thousands of files in a directory on an NFS mount can take a while and potentially block other actions). It's also much friendlier for migrating to an object store in the future.

Reply:
Maybe you could use directory names that are the first couple of characters from the uploads uuid. That's guaranteed not to change, and it is easier as a human to navigate the directories without having to know the created_at timestamp